### ISSUE TITLE
ThemeResource throws exception

### ISSUE THREAD

**Issue by Herdubreid** [microsoft/microsoft-ui-xaml#4686]
**Describe the bug**
When using ThemeResource SystemControlAltMediumLowAcrylicElementMediumBrush, the app crashes.
(Following https://github.com/windows-toolkit/MVVM-Samples/tree/master/samples)

**Steps to reproduce the bug**
Create new app and add a Border element to MainWindow.xaml using ThemeResource
```xaml
<Window
    x:Class="Test.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d">

    <Border Background="{ThemeResource SystemControlAltMediumLowAcrylicElementMediumBrush}" />
</Window>
```

**Expected behaviour**
App loads with selected background

**Version Info**

NuGet package version: 
Microsoft.ProjectReunion 0.5.0-prerelease

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (19042.867)              | Yes   |

**Additional context**

Copying the resource from generic.xaml to App.xaml (brush and color):
```xaml
<Application
    x:Class="Test.App"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
    <Application.Resources>
        <ResourceDictionary>
            <Color x:Key="SystemAltMediumLowColor">#66000000</Color>
            <SolidColorBrush x:Key="SystemControlAltMediumLowAcrylicElementMediumBrush" Color="{StaticResource SystemAltMediumLowColor}" />
            <ResourceDictionary.MergedDictionaries>
                <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" />
                <!-- Other merged dictionaries here -->
            </ResourceDictionary.MergedDictionaries>
            <!-- Other app resources here -->
        </ResourceDictionary>
    </Application.Resources>
</Application>
```
Works fine:
```xaml
<Window
    x:Class="Test.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d">
    
    <Border Background="{ThemeResource SystemControlAltMediumLowAcrylicElementMediumBrush}" />
</Window>
```

**Comment by StephenLPeters** [comment]
Is this a first chance execption that is handled upstream? (Can you f5 through the execption?)

**Comment by riverar** [comment]
@StephenLPeters Yep. Think this is same bug as #4496 (the infamous noise asset).

**Comment by codendone** [comment]
Also the same as #4510, which will be fixed in the upcoming 0.8 preview. That item can be used for further tracking.
